caddyconfig/httpcaddyfile/builtins.go

```diff
@@ -152,6 +152,9 @@ func parseTLS(h Helper) ([]ConfigValue, error) {
 			// policy that is looking for any tag but the last one to be
 			// loaded won't find it, and TLS handshakes will fail (see end)
 			// of issue #3004)
+
+			tlsCertTags := h.state["tlsCertTags"].(map[string]string)
```

Comment:
If this value doesn't exist (do an `, ok := ...` check), we should initialize and store it

Reply:
Sure ok, but I don't think there is another code path today that will mean this is uninitialised. Initialising at this point will mean we may load different certificates with the same tags. Would an error be better?

Reply:
Done